src/python/WMComponent/DBS3Buffer/DBSUploadPoller.py

```diff
@@ -110,14 +110,34 @@ def uploadWorker(workInput, results, dbsUrl, gzipEncoding=False):
                 logging.warning(msg)
                 results.put({'name': name, 'success': "error", 'error': msg})
             else:
-                msg = "Error trying to process block %s through DBS. Error: %s" % (name, exString)
+                reason = parseDBSException(exString)
+                msg = "Error trying to process block %s through DBS. Error: %s" % (name, reason)
                 logging.exception(msg)
                 logging.debug("block info: %s \n", block)
                 results.put({'name': name, 'success': "error", 'error': msg})
 
     return
 
 
+def parseDBSException(exc):
+    """
+    parse DBS Go-based server exception
+    :param exc: exception message
+    :return: either (parsed) concise exception message or original exception
+    """
+    try:
+        data = json.loads(exc)
+        # dbs2go always return a list
+        data = data[0]
```

Comment:
But response could be an empty list, right?
About the lines below, provided that it's not an empty list, can't we ensure that the response object will always have `error` and `reason` keys? This code looks safe, but I wish we didn't have to make all these ugly checks.

Reply:
Yes, this is what I meant. If Go server has a standard response structure, then we could remove all these checks:
```
        if 'error' in data:
            error = data['error']
            if 'reason' in error:
```
maybe just replace this whole code by something like:
```
try: 
    data = json.loads(exc)
    return data[0]['error']['reason']
except:
    return str(exc)
```
?